build: add --with-stage1-systemd-revision option for src build

s-urbaniak opened the pull request:
Currently the systemd version can be specified using
--with-stage1-systemd-version. The build system uses this value to
chcek out the systemd revision and uses the same value to set the
systemd version.

While this works nicely for building `vXXX` based tags, it causes
problems with custom systemd branch builds where an arbitrary branch
name does not map to a qualified systemd version.

Since rkt uses heuristics internally to evaluate the systemd version,
this causes conflicts.

This fixes it by introducing a new --with-stage1-systemd-revision
configure parameter to distinguish the systemd version, and revision
being built.

Fixes #3344

lucab commented:
Giving some time for @evverx to check this, postponing to next milestone.

evverx commented:
@s-urbaniak , thanks!

But I can (accidentally) omit the `--with-stage1-systemd-revision`:
```sh
# I extracted this from the bash history
./configure --enable-functional-tests --with-stage1-systemd-src=~/systemd --with-stage1-systemd-version=master --with-stage1-flavors=src
...
        systemd git revision:                   'master'
        systemd version:                        'master'
...
make
```
I think we should reject such versions.

BTW: we know
* the path to the repo
* the branch name

Why not to run something like
```sh
(git describe --contains BRANCH) || ($(git describe BRANCH) + 1)
```
?

s-urbaniak commented:
@evverx The path is known in your example, i.e. `--with-stage1-systemd-src=~/systemd`, but one can also specify `--with-stage1-systemd-src=https://github.com/systemd/systemd.git` (which is also the default).

So having a git-oneliner in the configure phase to determine the preceding/following tag name is not as easy as this patch; we need to check out the referenced repository first.

As good as this sounds I won't have time right now to dig deeper in the Makefile architecture to find the right spots at least to implement the requested enhanced version validation :-(

@lucab I added another small addition to the `build-and-run-tests.sh` script to actually also fix our `src_master` Jenkins build which will take a third parameter, namely the revision. The build matrix and the jenkins build script then only has to be changed for the src builds from `src_master` to `src_master_v999`, and from `src_v231` to `src_v231_v231` (I can do that once merged).

Finally I changed the defaults in the sec flavor build, `RKT_STAGE1_SYSTEMD_VER=v999`, and `RKT_STAGE1_SYSTEMD_REV=master`.

evverx commented:
@s-urbaniak , I think the `--with-stage1-systemd-revision=v...` is good enough. I use it and I'm happy :-)
Thanks!
